api/origin_trials_api.py

```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
+# Copyright 2023 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License")
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from framework import basehandlers
+from framework import secrets
+import requests
+import settings
+
+
+class OriginTrialsAPI(basehandlers.APIHandler):
+
+  def do_get(self, **kwargs):
+    """Get a list of all origin trials."""
+    key = secrets.get_ot_api_key()
+    # Return an empty list if no API key is found.
+    if key == None:
+      return []
+
+    try:
+      response = requests.get(
+          f'{settings.OT_API_URL}/v1/trials',
+          params={'prettyPrint': 'false', 'key': key})
+      response.raise_for_status()
+    except requests.exceptions.HTTPError:
+      self.abort(500, 'Error obtaining origin trial data from API.')
+
+    trials = response.json()['trials']
```

Comment:
I know it's low likelihood since we work on both OT API and ChromeStatus API now, but in the event the OT API ever changes and `trials` is missing, this line will throw an uncaught exception.

Technically, we should check if there's a trials key. If not, throw a 500 (since the server did not receive expected data to send back to the end user).

Reply:
Also, what is the exact data that we need to return back to the frontend? We should be explicit about the shape of the data we want to send back. That will prevent accidentally oversharing in case a field is added on the OT API side. Also, it will help out on the frontend to have the expected data response schema somewhere in this code.

Reply:
> Technically, we should check if there's a trials key. If not, throw a 500 (since the server did not receive expected data to send back to the end user).

Gotcha - I've added a check for this.

> Also, it will help out on the frontend to have the expected data response schema somewhere in this code.

Is this where OpenAPI comes in? I know we spoke about it, but I haven't familiarized myself with it yet. 😅 If this is the appropriate time, I'm willing to do so

Reply:
A dataclass will be sufficient. I think the plan is to use OpenAPI for the v1 of the API.